src/query.ts

```diff
@@ -204,231 +243,174 @@ export class Query {
     }
 
     (values as Array<unknown>).forEach((value: any) => {
-      if (this.hasExcludedFilter(filterName, value)) {
-        this.removeExcludedFilter(filterName, [value]);
+      if (this.hasExclusionFilter(filterName, value)) {
+        this.removeExclusionFilter(filterName, [value]);
       } else {
         this.addExcludeFilter(filterName, [value]);
       }
     });
   }
 
-  /**
-   * Sets the filter structure directly to the requests of a given context
-   *
-   * @param  {Object}     filters            The filter structure
-   *
-   * @param  {String}     filterType         If we are adding a "filter" (default)
-   *                                         or an "exclude" filter.
-   *
-   */
-  public setFilters(filters: Facet, filterType = 'filter'): void {
-    this.params[filterType] = filters;
-  }
-
-  /**
-   * Allows to directly set a parameter on the query builder
-   *
-   */
-  public setParameter(paramName: string, value: unknown): void {
-    if (paramName === 'sort') {
-      this.params['sort'] = this._parseSortingOptions(value as RequestSortOptions) as RequestSortOptions;
-    } else {
-      this.params[paramName] = value;
-    }
-  }
-
   /**
    * Overwrites the parameters with the object given, allowing
    * to change in one call several parameters
    *
    */
   public setParameters(parameters: SearchParameters): void {
     this.params = Object.assign({}, this.params, this._hydrate(parameters));
-  }
-
-  /**
-   * Sets an exclusion structure to the current context
-   *
-   * @param  {Object}        filters            The exclusion filter to add
-   *
-   */
-  public setExclusions(filters: Facet): void {
-    this.setFilters(filters, 'exclude');
-  }
-
-  /**
-   * Remove a sorting field from the parameters
-   *
-   * @param   {String}    field   The field we want to remove the ordering
-   *
-   */
-  public removeSorting(field: string): void {
-    if (!this.params.sort) {
-      return;
+    if ('sort' in this.params) {
+      this.addSort(this.params['sort']);
+      delete this.params['sort'];
     }
-
-    const sortParams = this.params.sort;
-    const index = this._findSortField(field, sortParams as Array<RequestSortOptions>);
-    if (index !== -1) {
-      (sortParams as Array<RequestSortOptions>).splice(index, 1);
-      if (sortParams.length === 0) {
-        delete this.params.sort;
-      } else {
-        this.params.sort = sortParams;
-      }
+    if ('rpp' in this.params) {
+      this.resultsPerPage(this.params['rpp']);
```

Comment:
I would name this `rpp` instead, it's much shorter and matches the parameter name.